Bump go version

acrespo opened the pull request.

acrespo committed:
Bump go version

acrespo committed:
Fix build problem during gomobile gobind exec

unable to import bind/java: [-: # golang.org/x/mobile/bind/java
cgo: C compiler "gcc" not found: exec: "gcc": executable file not found in $PATH]

acrespo committed:
Apollo: fix gomobile setup issue

acrespo committed:
Apollo: try full package fix

acrespo committed:
Apollo: run go get github.com/muun/libwallet

acrespo committed:
Apollo: run go get golang.org/x/mobile/cmd/gomobile@v0.0.0-20220414153400-ce6a79cf6a13

acrespo committed:
Apollo: new extra param is required in gomobile invocation

As per https://stackoverflow.com/a/77686965/901465.

Using api 19, our min sdk version.

menduz commented on libwallet/go.sum:
```diff
@@ -1,6 +1,7 @@
 cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
 cloud.google.com/go v0.33.1 h1:fmJQWZ1w9PGkHR1YL/P7HloDvqlmKQ4Vpb7PC2e+aCk=
 cloud.google.com/go v0.33.1/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
+dmitri.shuralyov.com/gpu/mtl v0.0.0-20221208032759-85de2813cf6b/go.mod h1:H6x//7gZCb22OMCxBHrMx7a5I7Hp++hsVxbQ4BYO7hU=
```
🫨

champo commented on android/Dockerfile:
```diff
@@ -13,6 +13,7 @@ RUN apt-get update \
         curl \
         git \
         zip \
+        build-essential\
```
we probably need to find a way to fix the gcc version. this will download "latest" which can differ between runs of the dockerfile and cause non-reproducible builds.

menduz replied:
<img width="484" alt="Screenshot 2024-07-09 at 12 38 04" src="https://github.com/acrespo/apollo/assets/260114/50f564aa-9d81-49ef-9e35-283c19c2ec2d">

acrespo replied:
We believe this is no longer needed, we introduced this to fix a compiling error before trying out the "full"/rebased changes from upstream. Those include a commit where they no longer try to compile certain files before hand. Will try removing this

acrespo committed:
Revert "Fix build problem during gomobile gobind exec"

This reverts commit 725a91e1c57c3ae3649227919ae9d8bb76a08984.

acrespo committed:
Run go mod tidy + go get golang.org/x/mobile/cmd/gomobile

Specifically:
go get golang.org/x/mobile/cmd/gomobile@v0.0.0-20220414153400-ce6a79cf6a13

acrespo committed:
Replace darbelo personal fork ref with muun official gombile fork

acrespo committed:
Rm go.sum no longer needed entries

acrespo committed:
Mv bootstrap-gomobile.sh to libwallet-android.sh

Simplifies env setup in other cases (e.g ci toolkit)
